src/librustc_mir/interpret/place.rs

```diff
@@ -1082,17 +1082,21 @@ where
                 }
                 if variant_index != dataful_variant {
                     let variants_start = niche_variants.start().as_u32();
-                    let variant_index_relative = variant_index.as_u32()
-                        .checked_sub(variants_start)
-                        .expect("overflow computing relative variant idx");
+
+                    let (variant_index_relative, op) = if variant_index.as_u32() >= variants_start {
```

Comment:
You can use overflowing_sub here to remove some duplication. The add/sub decision can be made on the overflow bool returned by overflowing_sub